wee_slack.py

```diff
@@ -60,97 +63,33 @@ def dbg(message, fout=False, main_buffer=False):
-        return string
-
-    def __repr__(self):
-        self.search_list.get_all(self.attribute)
-
-    def __getitem__(self, index):
-        things = self.get_all()
-        return things[index]
-
-    def __iter__(self):
-        things = self.get_all()
-        for channel in things:
-            yield channel
-
-    def get_all(self):
-        items = []
-        items += self.search_list.get_all(self.attribute)
-        return items
-
-    def find(self, name):
-        items = self.search_list.find_deep(name, self.attribute)
-        items = [x for x in items if x is not None]
-        if len(items) == 1:
-            return items[0]
-        elif len(items) == 0:
-            pass
-        else:
-            dbg("probably something bad happened with meta items: {}".format(items))
-            return items
-            #raise AmbiguousProblemError
-
-    def find_first(self, name):
-        items = self.find(name)
-        if items.__class__ == list:
-            return items[0]
-        else:
-            return False
-
-    def find_by_class(self, class_name):
-        items = self.search_list.find_by_class_deep(class_name, self.attribute)
-        return items
-
 
 class SearchList(list):
+    """
+    A normal python list with some syntactic sugar for searchability
+    """
+    def __init__(self):
+        self.hashtable = {}
+        super(SearchList, self).__init__(self)
 
     def find(self, name):
-        items = []
+        if name in self.hashtable.keys():
+            return self.hashtable[name]
+        #this is a fallback to __eq__ if the item isn't in the hashtable already
+        if self.count(name) > 0:
+            self.update_hashtable()
+            return self[self.index(name)]
+
+    def append(self, item, aliases=[]):
+        super(SearchList, self).append(item)
+        self.update_hashtable()
+
+    def update_hashtable(self):
         for child in self:
-            if child.__class__ == self.__class__:
-                items += child.find(name)
-            else:
-                if child == name:
-                    items.append(child)
-        if len(items) == 1:
-            return items[0]
-        elif items != []:
-            return items
-
-    def find_deep(self, name, attribute):
-        items = []
-        for child in self:
-            if child.__class__ == self.__class__:
-                if items is not None:
-                    items += child.find_deep(name, attribute)
-            elif dir(child).count('find') == 1:
-                if items is not None:
-                    items.append(child.find(name, attribute))
-        if items != []:
-            return items
-
-    def get_all(self, attribute):
-        items = []
-        for child in self:
-            if child.__class__ == self.__class__:
-                items += child.get_all(attribute)
-            else:
-                items += (eval("child." + attribute))
-        return items
+            if "get_aliases" in dir(child):
```

Comment:
I like `hasattr(child, "get_aliases") and callable(child.get_aliases)` better. `dir` is more of a debugging tool